config/application.rb

```diff
@@ -100,6 +100,26 @@ class Foreman::Consoletie < Rails::Railtie
 
 module Foreman
   class Application < Rails::Application
+    config.load_defaults '6.1'
+
+    # Rails 5.0 changed this to true, but a lot of code depends on this
+    config.active_record.belongs_to_required_by_default = false
+
+    # Rails 5.2 changed this to true, but we already do this in app/controllers/application_controller.rb#7
+    # We don't use this default because it's applied to ActionController::Base, thus to all inherited controllers
+    # But for API controllers we use a modification: app/controllers/concerns/foreman/controller/api_csrf_protection.rb#7
+    config.action_controller.default_protect_from_forgery = false
+    # Rails 5.2 changed this to true, but the only thing it does currently (as of Rails 7.0) is
+    # changing default cipher from aes-256-cbc to aes-256-gcm.
+    # Leaving this disabled, since the application worked with aes-256-cbc.
+    # Failed tests on aes-256-gcm require revisit application to ensure we can do the switch.
+    config.active_support.use_authenticated_message_encryption = false
```

Comment:
It's not a regression compared to today so we don't have to fix it right now. We should still fix it since cbc ciphers are considered weak nowadays so I'd be fine with opening a separate ticket for that.